ydb/public/lib/ydb_cli/commands/interactive/yql_highlight_ut.cpp

```diff
@@ -0,0 +1,172 @@
+        Check(highlight, "SELECT", "kkkkkk");
+        Check(highlight, "select", "kkkkkk");
+        Check(highlight, "ALTER", "kkkkk");
+        Check(highlight, "GROUP BY", "kkkkk kk");
+        Check(highlight, "INSERT", "kkkkkk");
+    }
+
+    Y_UNIT_TEST(Operation) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "(1 + 21 / 4)", "on o nn o no");
+        Check(highlight, "(1+21/4)", "ononnono");
+    }
+
+    Y_UNIT_TEST(FunctionIdentifier) {
+        YQLHighlight highlight(Coloring);
+
+        Check(highlight, "MIN", "fff");
+        Check(highlight, "min", "fff");
+        Check(highlight, "MIN(123, 65)", "fffonnno nno");
+        Check(highlight, "MIN", "fff");
+
+        Check(highlight, "minimum", "vvvvvvv");
+        Check(highlight, "MINimum", "vvvvvvv");
+
+        Check(highlight, "Math::Sin", "ffffoofff");
+        Check(highlight, "Math", "vvvv");
+        Check(highlight, "Math::", "ffffoo");
+        Check(highlight, "::Sin", "oovvv");
+    }
+
+    Y_UNIT_TEST(TypeIdentifier) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "Bool", "tttt");
+        Check(highlight, "Bool(value)", "ttttovvvvvo");
+    }
+
+    Y_UNIT_TEST(VariableIdentifier) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "test", "vvvv");
+    }
+
+    Y_UNIT_TEST(QuotedIdentifier) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "`/cluster/database`", "qqqqqqqqqqqqqqqqqqq");
+    }
+
+    Y_UNIT_TEST(String) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "\"\"", "ss");
+        Check(highlight, "\"test\"", "ssssss");
+        Check(highlight, "\"", "o");
+        Check(highlight, "\"\"\"", "sso");
+        Check(highlight, "\"\\\"", "ooo");
+        Check(highlight, "\"\\\"\"", "ssss");
+    }
+
+    Y_UNIT_TEST(Number) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "1234", "nnnn");
+        Check(highlight, "-123", "onnn");
+    }
+
+    Y_UNIT_TEST(SQL) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "SELECT id, alias from users",
+              "kkkkkk vvo vvvvv kkkk vvvvv");
+        Check(highlight, "INSERT INTO users (id, alias) VALUES (12, \"tester\")",
+              "kkkkkk kkkk vvvvv ovvo vvvvvo kkkkkk onno sssssssso");
+        Check(
+            highlight,
+            "SELECT 123467, \"Hello, {name}!\", (1 + (5 * 1 / 0)), MIN(identifier)"
+            "FROM `local/test/space/table` JOIN test;",
+            "kkkkkk nnnnnno sssssssssssssssso on o on o n o nooo fffovvvvvvvvvvo"
+            "kkkk qqqqqqqqqqqqqqqqqqqqqqqq kkkk vvvvo");
+        Check(highlight, "SELECT Bool(phone) FROM customer",
+              "kkkkkk ttttovvvvvo kkkk vvvvvvvv");
+    }
+
+    Y_UNIT_TEST(Emoji) {
+        YQLHighlight highlight(Coloring);
+        Check(highlight, "☺", "uuu");
+        Check(highlight, "\"☺\"", "sssuu");
+        Check(highlight, "`☺`", "qqquu");
```

Comment:
I don't know how it works, but it is highlighted normally:

![image](https://github.com/user-attachments/assets/5471a737-63f7-4ef7-9eeb-6cd2d8762416)